src/app/container/tool-file-editor/tool-file-editor.component.ts

```diff
@@ -77,16 +84,16 @@ export class ToolFileEditorComponent extends FileEditing {
     const message = 'Save Version';
     const combinedSourceFiles = this.getCombinedSourceFiles();
     const newSourceFiles = this.commonSaveVersion(this.originalSourceFiles, combinedSourceFiles);
-
+    this.alertService.start('Editing hosted tool');
     this.hostedService.editHostedTool(
         this.id,
         newSourceFiles).subscribe(result => {
           this.toggleEdit();
           this.containerService.setTool(result);
-          this.refreshService.handleSuccess(message);
+          this.alertService.detailedSuccess();
         }, error =>  {
           if (error) {
-            this.refreshService.handleError(message, error);
+            this.alertService.detailedError(error);
```

Comment:
What if `error` is not set (can that occur)? Do you still need to call alertService?

Reply:
not sure how there can be no error but if there no error, call simpleError()